File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/connection/HelloConnectMessage.scala





















// Generated on 05/03/2013 10:34:30
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait HelloConnectMessage extends MessageDeserializer {
  val messageId = 3

  protected trait Message extends BaseMessage with Serializable {
    val messageId = HelloConnectMessage.messageId

    val salt: String
    val key: Seq[Byte]

    def serialize(buf: Buffer) {
      buf.writeString(salt)
      buf.writeUShort(key.length)
      for (entry <- key) {
        buf.writeByte(entry)
      }
    }

  }

  protected def create(salt: String, key: Seq[Byte]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val salt = buf.readString()
    val key = {
      val builder = collection.immutable.Seq.newBuilder[Byte]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readByte()
      }
      builder.result()
    }

    create(salt, key)
  }

}

object HelloConnectMessage extends HelloConnectMessage {
  case class MessageImpl(salt: String, key: Seq[Byte]) extends Message
  type Target = MessageImpl

  override def create(salt: String, key: Seq[Byte]) = MessageImpl(salt, key)

  def apply(salt: String, key: Seq[Byte]) = create(salt, key)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/HelloConnectMessage.java


// Generated on 05/08/2013 19:37:37
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class HelloConnectMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 3;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public String salt;
    public byte[] key;
    
    public HelloConnectMessage() { }
    
    public HelloConnectMessage(String salt, byte[] key) {
        this.salt = salt;
        this.key = key;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeString(salt);
        buf.writeUShort(key.length);
        for (byte entry : key) {
            buf.writeByte(entry);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        salt = buf.readString();
        int limit = buf.readUShort();
        key = new byte[limit];
        for (int i = 0; i < limit; i++) {
            key[i] = buf.readByte();
        }
    }
    
}
